test/implementations/test_mct_barenco_74_dirty.py

```diff
@@ -56,6 +57,14 @@ def _take_auxiliaries_no(self, controls_no: int):
 
         return self._auxiliary_dict[controls_no]
 
+    def test_init(self):
+        with pytest.raises(Exception) as e_info:
+            MCTBarenco74Dirty(2)
+
+        assert e_info.value.args[0] == "Number of controls must be >= 5 for this implementation"

```

Comment:
check `match` in https://docs.pytest.org/en/7.1.x/how-to/assert.html

Reply:
```
with pytest.raises(ValueError, match="Number of controls must be >= 5 for this implementation"):
    MCTBarenco74Dirty(2)
```

Reply:
fixed with the last commit